eogrow/utils/validators.py

```diff
@@ -214,12 +214,13 @@ def parse_data_collection(value: Union[str, dict, DataCollection]) -> DataCollec
     return DataCollection.define(name, **params)
 
 
-def restrict_types(allowed_feature_types: Iterable[FeatureType]) -> Callable[[Feature], Feature]:
+def restrict_types(
+    allowed_feature_types: Union[Iterable[FeatureType], Callable[[FeatureType], bool]]
+) -> Callable[[Feature], Feature]:
     """Validates a field representing a feature, where it restricts the possible feature types."""
 
     def validate_feature(value: Feature) -> Feature:
-        ftype, _ = value
-        assert ftype in allowed_feature_types, f"Feature type can be one of {allowed_feature_types}, got {ftype}."
+        FeatureParser(features=value, allowed_feature_types=allowed_feature_types)
```

Comment:
it feel like we're relying on the current implementation of `FeatureParser` to raise the issue on `__init__`. Perhaps if we go with `parse_feature` directly?

Reply:
Agreed. Updated.